Configurable x-pack-security settings

dliappis opened the pull request:
This commit add support for configuring the user name, password and
role for the Elasticsearch (file realm) user generated by Rally when
using the x-pack-security car.

Also adjust docs now that x-pack-security is a car and not a plugin.

danielmitterdorfer commented on cars/v1/x_pack/README.md:
```diff
@@ -1,18 +1,29 @@
-* `--elasticsearch-plugins="x-pack:security"`
-* `--elasticsearch-plugins="x-pack:monitoring-local"`
-* `--elasticsearch-plugins="x-pack:security+monitoring-local"`
+* `--car-plugins="x-pack-security"`
```
* The command line parameter says `--car-plugins` but should be `--car`
* All of these are mixins so they require a "proper" car, i.e. it should be `--car="defaults,x-pack-security"`

danielmitterdorfer commented on cars/v1/x_pack/README.md:
```diff
@@ -1,18 +1,29 @@
-This directory contains example configurations for x-pack:
+# This directory contains example configurations for x-pack:
```
To me this sounds odd for a heading? How about "Provided configurations"? Also, for other plugins like `repository-gcs` we don't have a heading at all so maybe just drop it?

danielmitterdorfer commented on cars/v1/x_pack/README.md:
```diff
@@ -1,18 +1,29 @@
 
 * `security`: Configures TLS for all HTTP and transport communication using self-signed certificates.
-* `monitoring-local`: Enables x-pack monitoring with export to the current cluster.
+* `monitoring`: Enables x-pack monitoring with export to the current cluster.
```
Should this be `monitoring-http` and the description should be changed in the sense that it exports to a different cluster? Also, `ml` seems to be missing.

danielmitterdorfer commented on cars/v1/x_pack/README.md:
```diff
@@ -1,18 +1,29 @@
+* `--car-plugins="x-pack-security,x-pack-monitoring-local"`
 
-The `security` configuration will enable basic authentication and TLS for the HTTP and the transport layer. It will also add a `rally` super-user with the password `rally-password`. As a consequence, you will need to add the following client options when benchmarking this configuration:
+## Configuring security user name, password, role
```
In `repository-gcs` we've called this section `Parameters`. Maybe we should align it?

dliappis commented:
@danielmitterdorfer I've addressed the above comments (and expanded the README to provide information for the properties exposed by the rest of the x-pack cars). WDYT?
